### PR TITLE
Automatic plugin loading

### PR DESCRIPTION
**Pull request by suneettipirneni**
This allows for automatic plugin loading from the `plugins/` folder. This removes the previously needed step of adding your plugin manually to the `plugin.loader.ts`. 

Notable changes:
- new `Plugin` property, `commandName`, this property is the base command name the plugin uses.
- In order for plugins to be automatically loaded, they will need to be default exports.

### CODE REVIEW

File: src/app/bot.ts [hunk 1]
```diff
@@ -36,18 +36,33 @@ export class Bot {
...
+            this.container.loggerService.error(`${file} has a default export, but it is not of type Plugin`);
+            return;
+          }
+        } catch(err) {
+          this.container.loggerService.warn(`${file} doesn't have a default export of type Plugin!`);
+          return;
+        }
+
+        // Register plugin.
+        this.container.pluginService.register(plugin);
```
Comment (joey-colon) [inline src/app/bot.ts:63]: can we move this to be within the try? and then remove the `return` within the error block

File: src/app/bot.ts [hunk 2]
```diff
@@ -36,18 +36,33 @@ export class Bot {
...
-        .forEach((plugin) => this.container.pluginService.register(plugin, this.container));
-    } catch (e) {
-      this.container.loggerService.error(e);
-    }
+    const pluginFolder = './src/app/plugins';
+    fs.readdir(pluginFolder, (_err, files) => {
+      files.forEach(async file => {
+
+        // Import the class from the plugin file.
+        const thing = await import(`./plugins/${file}`);
```
Comment (joey-colon) [inline src/app/bot.ts:44]: can we use a more descriptive name? maybe `pluginImport` or rename what `plugin` is known as now to `pluginInstance` and move `thing` to be = `plugin`

**Review by tanndlin — COMMENTED** [review]
Is creating a display name relevant to this topic?

File: src/common/plugin.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ export abstract class Plugin implements IPlugin {
 
   public abstract get name(): string;
```
Comment (joey-colon) [inline src/common/plugin.ts:7]: i think this property should still be referring to the plugin name. to have it refer to the command name is kind of confusing when we consider the base class and interface we're abiding to is called `Plugin`.

can we maybe refer to `name` as the name of plugin and change `displayName` to be `command` or `commandName`?

File: src/app/bot.ts (hunk 2 above)
Comment (suneettipirneni) [reply src/app/bot.ts:44]: 👀

**Comment by suneettipirneni** [comment]
> Is creating a display name relevant to this topic?

The `displayName` property is just the new name for the older `name` property.

**Comment by tanndlin** [comment]
> > Is creating a display name relevant to this topic?
> 
> The `displayName` property is just the new name for the older `name` property.

Oh I see what you did now

**Review by tanndlin — APPROVED** [review]

**Review by Armintron — APPROVED** [review]
Looks good to me

**Pull request merged**